src/UsbPrinterDeviceChannel.ts

```diff
@@ -0,0 +1,134 @@
+import { IPrinterDeviceChannel } from './IPrinterDeviceChannel';
+import { LineBreakTransformer } from './LineBreakTransformer';
+import { PrinterCommunicationMode } from './PrinterCommunicationMode';
+
+/**
+ * Class for managing the WebUSB communication with a printer.
+ */
+export class UsbPrinterDeviceChannel extends EventTarget implements IPrinterDeviceChannel {
+    private device: USBDevice;
+    private deviceIn: USBEndpoint;
+    private deviceOut: USBEndpoint;
+
+    public EnableConsoleDebug = false;
+
+    private commMode: PrinterCommunicationMode;
+    /**
+     * Gets the mode the communication is set up as.
+     */
+    public CommMode() {
+        return this.commMode;
+    }
+
+    private ready: Promise<boolean>;
+    public Ready(): Promise<boolean> {
+        return this.ready;
+    }
+
+    private inputStream: ReadableStream;
+    public StreamFromPrinter() {
+        return this.inputStream;
+    }
+
+    constructor(device: USBDevice, enableConsoleDebug = false) {
+        super();
+
+        this.device = device;
+        this.EnableConsoleDebug = enableConsoleDebug;
+
+        this.ready = new Promise((resolve, reject) => {
```

Comment:
This could also just be 
```this.ready = this.Connect();```

Reply:
That's the point of the Ready promise, generally. More conceptually I don't believe there's a compelling reason to have a channel that isn't open, that is, I couldn't come up with a strong reason to allow users of the library to separate creating a printer from connecting to a printer.

The vibe I was getting off some other libraries was that having a constructor and then expecting everyone to then call another method to set it up leads to issues with uninitialized objects being accessed or race conditions. By always firing off the async operation and then providing the hook point to await or chain onto was offered as a more robust solution for this sort of thing.

Exposing that promise externally also gives the hook point for receiving useful information, such as errors.